### PR TITLE
FT2023-342: Addition of all the files of the route module.

### PR DESCRIPTION
**Pull request by Arijit-Acharya-innoraft**

### CODE REVIEW

File: web/modules/custom/myroute/myroute.permissions.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+view myroute:
+  title: View my module.
+  description: View the my module module's content.
```
Comment (vivekpan27) [inline web/modules/custom/myroute/myroute.permissions.yml]: Please encapsulate title and description values in single quotes.

File: web/modules/custom/myroute/myroute.routing.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+myroute.practicingRoute:
+  path: '/routeit/practice'
+  defaults: 
+    _controller: 'Drupal\myroute\Controller\MyrouteController::practicingRoute'
+    _title: 'Route Content'
+  requirements:
+    _permission: 'access custom page'
+    _role: administrator + 	content_editor
+
```
Comment (vivekpan27) [inline web/modules/custom/myroute/myroute.routing.yml:9]: unnecessary extra space

File: web/modules/custom/myroute/myroute.routing.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+myroute.practicingRoute:
```
Comment (vivekpan27) [inline web/modules/custom/myroute/myroute.routing.yml]: `myroute.practicing_route:`

Please change the other route ids similarly.

File: web/modules/custom/myroute/myroute.routing.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  defaults: 
+    _controller: 'Drupal\myroute\Controller\MyrouteController::practicingRoute'
+    _title: 'Route Content'
+  requirements:
+    _permission: 'access custom page'
+    _role: administrator + 	content_editor
+
+
+myroute.getArticleNo:
+  path: '/campaign/value/{id}'
```
Comment (vivekpan27) [inline web/modules/custom/myroute/myroute.routing.yml:11]: Please add proper validation on the id paramater.

File: web/modules/custom/myroute/src/Controller/MyrouteController.php [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+  /**
+   * Displays a message on the practicing route.
+   *
+   * @return array
+   *   A render array containing the markup for the message.
+   */
+  public function practicingRoute() {
+    if ($this->account->hasPermission('view myroute')) {
+      AccessResult::Allowed();
```
Comment (vivekpan27) [inline web/modules/custom/myroute/src/Controller/MyrouteController.php]: What are we trying to do here and in line 55?
We are not returning or using these in anyway.
So what's the purpose?

File: web/modules/custom/myroute/src/Controller/MyrouteController.php [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        '#markup' => 'hello! route',
+      ];
+    }
+    AccessResult::Forbidden();
+  }
+
+  /**
+   * Displays the article number.
+   *
+   * @param mixed $id
```
Comment (vivekpan27) [inline web/modules/custom/myroute/src/Controller/MyrouteController.php]: why mixed?

File: web/modules/custom/myroute/myroute.permissions.yml (hunk 1 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/myroute/myroute.permissions.yml]: Done

File: web/modules/custom/myroute/myroute.routing.yml (hunk 2 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/myroute/myroute.routing.yml:9]: Extra Space removed

File: web/modules/custom/myroute/myroute.routing.yml (hunk 3 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/myroute/myroute.routing.yml]: Changed.

File: web/modules/custom/myroute/src/Controller/MyrouteController.php (hunk 6 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/myroute/src/Controller/MyrouteController.php]: Was a typo. Updated.

File: web/modules/custom/myroute/src/Controller/MyrouteController.php (hunk 5 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/myroute/src/Controller/MyrouteController.php]: Have cross checked again, found that it is blocking the unauthorised users. 
However Updated by adding the return statements.

File: web/modules/custom/myroute/myroute.routing.yml (hunk 4 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/myroute/myroute.routing.yml:11]: added

File: web/modules/custom/myroute/myroute.routing.yml (hunk 2 above)
Comment (vivekpan27) [reply web/modules/custom/myroute/myroute.routing.yml:9]: There is a 2 line gap.
line 9,10

Is that necessary?

File: web/modules/custom/myroute/src/Controller/MyrouteController.php [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Defines a controller for the Myroute module.
+ */
+class MyrouteController extends ControllerBase {
+
+  /**
+   * The account object for the currently logged-in user.
```
Comment (vivekpan27) [inline web/modules/custom/myroute/src/Controller/MyrouteController.php]: It can't be for logged out user?

Comment (vivekpan27) [reply web/modules/custom/myroute/src/Controller/MyrouteController.php]: Just write User account service or something.

File: web/modules/custom/myroute/src/Controller/MyrouteController.php [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+   * Displays the article number.
+   *
+   * @param string $id
+   *   The ID of the article.
+   *
+   * @return array
+   *   A render array containing the markup for the article number.
+   */
+  public function getArticleNo($id) {
+
```
Comment (vivekpan27) [inline web/modules/custom/myroute/src/Controller/MyrouteController.php]: unnecessary extra line

Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/myroute/src/Controller/MyrouteController.php]: Removed

File: web/modules/custom/myroute/myroute.routing.yml (hunk 2 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/myroute/myroute.routing.yml:9]: Provided for better readability.
Removed as per the suggestion.

File: web/modules/custom/myroute/src/Controller/MyrouteController.php (hunk 7 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/myroute/src/Controller/MyrouteController.php]: Changed as per the suggestion.

**Comment by vivekpan27** [comment]
@Arijit-Acharya-innoraft Please take care that module names should be camel cased.
For this case, it should have been `my_route`.
But I'm letting it remain for now, please don't repeat it again.

**Pull request merged**